src/main/java/net/minecraftforge/registries/FieldObjectHolderRef.java

```diff
@@ -22,34 +22,27 @@
 import java.lang.reflect.Field;
 import java.lang.reflect.Method;
 import java.lang.reflect.Modifier;
-import java.util.Collections;
-import java.util.LinkedList;
-import java.util.Queue;
-
+import javax.annotation.Nullable;
 import net.minecraft.util.ResourceLocation;
-
 import net.minecraftforge.fml.common.FMLLog;
-import net.minecraftforge.fml.common.registry.GameRegistry;
-import net.minecraftforge.fml.common.registry.GameRegistry.ObjectHolder;
-
-import javax.annotation.Nullable;
 
 /**
- * Internal class used in tracking {@link ObjectHolder} references
+ * Default, field-based implementation for {@link IObjectHolderRef}.
  */
-@SuppressWarnings("rawtypes")
-class ObjectHolderRef
+public class FieldObjectHolderRef implements IObjectHolderRef
```

Comment:
Why make this a public class if the only constructor is package private still? And why change the name/constructor arguments if they are only package private?